test/api/unit/services/S3SiteRemover.test.js

```diff
@@ -1,82 +1,124 @@
+        cb(null, {});
+      };
 
-      factory.site().then(model => {
-        site = model
+      factory.site().then((model) => {
+        site = model;
 
-        const sitePrefix = `site/${site.owner}/${site.repository}`
-        siteObjectsToDelete.push(`${sitePrefix}/index.html`)
-        siteObjectsToDelete.push(`${sitePrefix}/redirect`)
-        siteObjectsToDelete.push(`${sitePrefix}/redirect/index.html`)
-        const demoPrefix = `demo/${site.owner}/${site.repository}`
-        demoObjectsToDelete.push(`${demoPrefix}/index.html`)
-        demoObjectsToDelete.push(`${demoPrefix}/redirect`)
-        demoObjectsToDelete.push(`${demoPrefix}/redirect/index.html`)
-        const previewPrefix = `preview/${site.owner}/${site.repository}`
-        previewObjectsToDelete.push(`${previewPrefix}/index.html`)
-        previewObjectsToDelete.push(`${previewPrefix}/redirect`)
-        previewObjectsToDelete.push(`${previewPrefix}/redirect/index.html`)
+        const sitePrefix = `site/${site.owner}/${site.repository}`;
+        siteObjectsToDelete.push(`${sitePrefix}/index.html`);
+        siteObjectsToDelete.push(`${sitePrefix}/redirect`);
+        siteObjectsToDelete.push(`${sitePrefix}/redirect/index.html`);
+        const demoPrefix = `demo/${site.owner}/${site.repository}`;
+        demoObjectsToDelete.push(`${demoPrefix}/index.html`);
+        demoObjectsToDelete.push(`${demoPrefix}/redirect`);
+        demoObjectsToDelete.push(`${demoPrefix}/redirect/index.html`);
+        const previewPrefix = `preview/${site.owner}/${site.repository}`;
+        previewObjectsToDelete.push(`${previewPrefix}/index.html`);
+        previewObjectsToDelete.push(`${previewPrefix}/redirect`);
+        previewObjectsToDelete.push(`${previewPrefix}/redirect/index.html`);
 
-        return S3SiteRemover.removeSite(site)
+        return S3SiteRemover.removeSite(site);
       }).then(() => {
-        expect(siteObjectsWereListed).to.equal(true)
-        expect(previewObjectsWereListed).to.equal(true)
-        expect(objectsWereDeleted).to.equal(true)
-        done()
-      }).catch(done)
-    })
-  })
-})
+        expect(siteObjectsWereListed).to.equal(true);
+        expect(demoObjectWereListed).to.equal(true);
+        expect(previewObjectsWereListed).to.equal(true);
+        expect(objectsWereDeleted).to.equal(true);
+        done();
+      }).catch(done);
+    });
+
+    it('should delete objects in batches of 1000 at a time', (done) => {
+      let deleteObjectsCallCount = 0;
+
+      AWSMocks.mocks.S3.listObjects = (params, cb) => cb(null, {
+        Contents: Array(750).fill(0).map(() => ({ Key: 'abc123' })),
+      });
+
+      AWSMocks.mocks.S3.deleteObjects = (params, cb) => {
+        expect(params.Delete.Objects).to.have.length.at.most(1000);
+        deleteObjectsCallCount += 1;
+        cb();
+      };
+
+      factory.site()
+      .then(site => S3SiteRemover.removeSite(site))
+      .then(() => {
+        expect(deleteObjectsCallCount).to.equal(3);
```

Comment:
Seeing the `750` above and the the `3` here kind of threw me off. An explanatory comment about why (from reading the code it is because of real, demo, and preview sites each having the 750 objects) mocking `listObjects` to return 750 elements will result in more than a single call to `deleteObjects` would be nice.